pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -93,26 +124,38 @@ func (gce *GCECloud) AttachDisk(diskName string, nodeName types.NodeName, readOn
 	if err != nil {
 		return fmt.Errorf("error getting instance %q", instanceName)
 	}
-	disk, err := gce.getDiskByName(diskName, instance.Zone)
-	if err != nil {
-		return err
+
+	// Try fetching as regional PD
+	var disk *GCEDisk
+	if gce.AlphaFeatureGate.Enabled(GCEDiskAlphaFeatureGate) {
+		disk, err = gce.getRegionalDiskByName(diskName)
+		if err != nil {
+			glog.Warningf("Could not find regional PD named %q to Attach. Will look for a zonal PD", diskName)
+			err = nil
+		}
+	}
+
+	if disk == nil {
+		disk, err = gce.getDiskByName(diskName, instance.Zone)
+		if err != nil {
+			return err
+		}
 	}
+
 	readWrite := "READ_WRITE"
 	if readOnly {
 		readWrite = "READ_ONLY"
 	}
 
-	mc := newDiskMetricContext("attach", instance.Zone)
-	source := gce.service.BasePath + strings.Join([]string{
-		gce.projectID, "zones", disk.Zone, "disks", disk.Name}, "/")
+	mc := newDiskMetricContextZonal("attach", gce.region, instance.Zone)
```

Comment:
Unsure if this should be newDiskMetricContextRegional if the disk is regional (not a blocker to a merge IMO)

Reply:
Fixed